### PR TITLE
Implement Delivery methods

### PR DESCRIPTION
**Pull request by agrawalarpit14**
Three types of delivery services are added in the PR. The service has to be chosen while creating an order on the route `/#/delivery-method`:
1.  name: 'One Day Delivery', price: 0.99, primeCustomerPrice: 0.5, eta: 1
2. name: 'Fast Delivery', price: 0.5, primeCustomerPrice: 0, eta: 3
3. name: 'Standard Delivery', price: 0, primeCustomerPrice: 0, eta: 5

### CODE REVIEW

**Comment by agrawalarpit14** [comment]
Hi @bkimminich @J12934 Please let me know your review on the PR.
Thank You!! so much.😁😁

**Comment by bkimminich** [comment]
Please reuse the `LABEL_DELIVERY_ADDRESS` string and term _Delivery Address_ instead of creating a new term of _Shipping Address_.

**Comment by agrawalarpit14** [comment]
Hi @bkimminich Sorry, for this one. I have corrected what you pointed out and from now on will make sure that `i18n` remains consistent.
Thank You!! so much

File: data/datacreator.js [hunk 1]
```diff
@@ -97,6 +98,25 @@ async function createUsers () {
...
+    delivery.map(async ({ name, price, primePrice, eta }) => {
+      try {
+        await models.Delivery.create({
+          name,
+          price,
+          primePrice,
+          eta
+        })
+      } catch (err) {
+        logger.error(`Could not insert User: ${err.message}`)
```
Comment (J12934) [inline data/datacreator.js]: Error message is incorrect here 😉

File: frontend/src/app/Services/delivery.service.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+})
+export class DeliveryService {
+
+  private hostServer = environment.hostServer
+  private host = this.hostServer + '/api/Deliverys'
+
+  constructor (private http: HttpClient) { }
+
+  get () {
+    return this.http.get(this.host).pipe(map((response: any) => response.data), catchError((err) => { throw err }))
```
Comment (J12934) [inline frontend/src/app/Services/delivery.service.ts]: The response observable could / should maybe also be typed.

Having types for API responses can be really helpful.

File: frontend/src/app/order-completion/order-completion.component.ts [hunk 3]
```diff
@@ -27,9 +27,10 @@ export class OrderCompletionComponent implements OnInit {
       this.orderId = paramMap.get('id')
       this.trackOrderService.save(this.orderId).subscribe((results) => {
         this.promotionalDiscount = results.data[0].promotionalAmount ? parseFloat(results.data[0].promotionalAmount) : 0
+        this.deliveryPrice = results.data[0].deliveryPrice ? parseFloat(results.data[0].deliveryPrice) : 0
         this.orderDetails.addressId = results.data[0].addressId
         this.orderDetails.paymentId = results.data[0].paymentId
-        this.orderDetails.totalPrice = results.data[0].totalPrice + this.promotionalDiscount
+        this.orderDetails.totalPrice = results.data[0].totalPrice + this.promotionalDiscount - this.deliveryPrice
```
Comment (J12934) [inline frontend/src/app/order-completion/order-completion.component.ts]: Havent' had much time to read through this but subtracting the delivery price seems weird to me. Is this correct like this?

Shouldn't this be `+ delivery price - promotionalDiscount`?

Comment (agrawalarpit14) [reply frontend/src/app/order-completion/order-completion.component.ts]: Actually, on the backend `totalPrice` is supposed to be the price user pays. So, on the backend `totalPrice = itemTotal + deliveryPrice - promotionalDiscount`. As a result, to get back itemTotal on the frontend, I need to add `promotionalDiscount` and subtract back `deliveryPrice `.😅
I have made the naming more consistent to make it clear.😁😁

File: data/datacreator.js (hunk 1 above)
Comment (agrawalarpit14) [reply data/datacreator.js]: My bad😅

File: frontend/src/app/Services/delivery.service.ts (hunk 2 above)
Comment (agrawalarpit14) [reply frontend/src/app/Services/delivery.service.ts]: I have added the types, but I was wondering if we could use [GraphQL](https://graphql.org) in our project??

Comment (bkimminich) [reply frontend/src/app/Services/delivery.service.ts]: GraphQL was an idea quite some time ago, but was dropped back then, see #330. If a really good idea comes up for a use case, I'm happy to add it, but not "on the fly" ... :-)

Comment (agrawalarpit14) [reply frontend/src/app/Services/delivery.service.ts]: Sure, I will think of something.😁😁

**Comment by agrawalarpit14** [comment]
Hi @bkimminich I have updated this PR including the changes `pincode` --> `zipCode` and the currency symbol. Please have a look.
Thank You so much😁😁

**Pull request merged**